psutil/_psosx.py

```diff
@@ -212,6 +213,42 @@ def disk_partitions(all=False):
 # =====================================================================
 
 
+def sensors_temperatures():
+    """Return CPU temperatures
+    including key name and temperature.
+    """
+    ret = collections.defaultdict(list)
+
+    # This handles returning per-core temperature. The SMCkeys for core
+    # temperatue are known
+    rawlist = cext.cpu_cores_temperatures()
+    for core in rawlist:
+        ret["CPU"].append((core[0], core[1], None, None))
+
+    # Additional CPU sensors
+    rawlist = cext.cpu_misc_temperatures()
+    for core in rawlist:
+        ret["CPU"].append((core[0], core[1], None, None))
+
+    try:
+        rawlist = cext.battery_temperatures()
+        if rawlist is not None:
+            for batt in rawlist:
+                ret["Batteries"].append((batt[0], batt[1], None, None))
+    except(SystemError):
```

Comment:
Why? In general you want to let the underlying C extension raise exceptions (e.g. `OSError`). In this case if `cext.battery_temperatures()` fails because the SMC key does not exist then the C extension should simply return an empty list or dict